test/objectreference.js

```diff
@@ -0,0 +1,255 @@
+      const test2 = new Array();
+      test2[0] = binding.objectreference.getCastedFromGetter("persistent", 0);
+      test2[1] = binding.objectreference.getCastedFromGetter("persistent", 1);
+      test2[2] = binding.objectreference.getCastedFromGetter("persistent", 2);
+    
+      assert.ok(test instanceof Array);
+      assert.ok(test2 instanceof Array);
+      testCastedEqual(test);
+      testCastedEqual(test2);
+    },
+
+    'References Casted Array',
+    () => {
+      binding.objectreference.setCastedObjects();
+      const test = binding.objectreference.getCastedFromValue();
+      const test2 = new Array();
+      test2[0] = binding.objectreference.getCastedFromGetter("reference", 0);
+      test2[1] = binding.objectreference.getCastedFromGetter("reference", 1);
+      test2[2] = binding.objectreference.getCastedFromGetter("reference", 2);
+    
+      assert.ok(test instanceof Array);
+      assert.ok(test2 instanceof Array);
+      testCastedEqual(test);
+      testCastedEqual(test2);
+    },
+
+    'Weak',
+    () => {
+      binding.objectreference.setObjects("hello", "world");
+      const test = binding.objectreference.getFromValue("weak");
+      const test2 = binding.objectreference.getFromGetter("weak", "hello");
+
+      assert.deepEqual({ hello: "world"}, test);
+      assert.equal("world", test2);
+      assert.equal(test["hello"], test2);
+    },
+    () => {
+      binding.objectreference.setObjects("hello", "world", "javascript");
+      const test = binding.objectreference.getFromValue("weak");
+      const test2 = binding.objectreference.getFromValue("weak", "hello");
+
+      assert.deepEqual({ hello: "world" }, test);
+      assert.deepEqual({ hello: "world" }, test2);
+      assert.equal(test, test2);  
+    },
+    () => {
+      binding.objectreference.setObjects(1, "hello world");
+      const test = binding.objectreference.getFromValue("weak");
+      const test2 = binding.objectreference.getFromGetter("weak", 1);
+
+      assert.deepEqual({ 1: "hello world" }, test);
+      assert.equal("hello world", test2);
+      assert.equal(test[1], test2);
+    },
+    () => {
+      binding.objectreference.setObjects(0, "hello");
+      binding.objectreference.setObjects(1, "world");
+      const test = binding.objectreference.getFromValue("weak");
+      const test2 = binding.objectreference.getFromGetter("weak", 0);
+      const test3 = binding.objectreference.getFromGetter("weak", 1);
+
+      assert.deepEqual({ 1: "world" }, test);
+      assert.equal(undefined, test2);
+      assert.equal("world", test3);
+    },
+    () => {
+      binding.objectreference.setObjects("hello", "world");
+      assert.throws(
+        () => {
+          var rcount = binding.objectreference.refObjects("weak");
+          assert.equal(rcount, 1);
+          rcount = binding.objectreference.unrefObjects("weak");
+          assert.equal(rcount, 0);
+          binding.objectreference.unrefObjects("weak");
+        },
+        Error
```

Comment:
If the first 2 asserts are supposed to pass I'm not sure this would catch failures of those asserts ?  (Since they throw and overall the assert.throws does not check what Error occured ?